### PR TITLE
API Generator: Add missing type for integer fields in input type

### PR DESCRIPTION
**Pull request by Ben-Ho**
API-Generator should generate `@Field(() => Int, ...)` in input-type for an entity-property with integer column-type.

this
```
@ObjectType()
@Entity()
@CrudGenerator({ targetDirectory: `${__dirname}/../generated/`, requiredPermission: ["products"] })
export class MyEntity extends BaseEntity<MyEntity, "id"> {
    ...
    @Property({ columnType: "int" })
    @Field(() => Int)
    position: number;
    ...
}
```

results in 
```
@InputType()
export class MyEntityInput {
    ....
    @IsNotEmpty()
    @IsInt()
    @Field()
    position: number;
}
```
but should generate

```
@InputType()
export class MyEntityInput {
    ....
    @IsNotEmpty()
    @IsInt()
    @Field(() => Int)
    position: number;
}
```
for the schema.gql to correctly include Int-type instead of Float!
```
input MyEntityInput {
  ...
  position: Int!
}
```

### CODE REVIEW

File: packages/api/cms-api/src/generator/generate-crud-input.ts [hunk 1]
```diff
@@ -118,12 +118,14 @@ export async function generateCrudInput(
...
             const fieldOptions = tsCodeRecordToString({ nullable: prop.nullable ? "true" : undefined, defaultValue });
+            let returnTypeFunction = ``;
             if (integerTypes.includes(prop.columnTypes[0])) {
                 decorators.push("@IsInt()");
+                returnTypeFunction = `() => Int, `;
             } else {
                 decorators.push("@IsNumber()");
             }
-            decorators.push(`@Field(${fieldOptions})`);
+            decorators.push(`@Field(${returnTypeFunction}${fieldOptions})`);
```
Comment (Ben-Ho) [inline packages/api/cms-api/src/generator/generate-crud-input.ts:128]: I tried to update the generate-crud-input-integer.spec.ts but I found the decorator arguments fields empty.

File: .changeset/shy-jars-float.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+"@comet/cms-api": patch
+---
+
+Fix api-generator missing return-type function for Int fields in input-type
```
Comment (johnnyomair) [inline .changeset/shy-jars-float.md:5]:
```suggestion
API Generator:  Add missing return type function for integer fields in input type
```

File: .changeset/shy-jars-float.md [hunk 3]
```diff
@@ -0,0 +1,5 @@
+---
+"@comet/cms-api": patch
+---
+
+API Generator:  Add missing return type function for integer fields in input type
```
Comment (johnnyomair) [inline .changeset/shy-jars-float.md:5]:
```suggestion
API Generator: Add missing return type function for integer fields in input type
```

File: .changeset/shy-jars-float.md [hunk 4]
```diff
@@ -0,0 +1,5 @@
+---
+"@comet/cms-api": patch
+---
+
+API Generator: Add missing return type function for integer fields in input type
```
Comment (nsams) [inline .changeset/shy-jars-float.md:5]:
```suggestion
API Generator: Add missing type for integer fields in input type
```

**Review by nsams — APPROVED** [review]

**Review by johnnyomair — APPROVED** [review]

**Pull request merged**